core/templates/dev/head/pages/exploration_editor/editor_tab/ExplorationEditorTab.js

```diff
@@ -152,6 +152,28 @@ oppia.controller('ExplorationEditorTab', [
         $scope.stateName, angular.copy(displayedValue));
     };
 
+    $scope.showMarkAllAudioAsNeedingUpdateModalIfRequired = function(
```

Comment:
I like how pulling this out into a separate place reduces quite a lot of repeated code. Good move!

The only thing I would suggest is that this is moved into its own service so that you don't have to thread the dependency all the way down the chain of components. This is also in line with the principle that controllers should be thin.

Reply:
@seanlip This needs a thought to work on as the state structure is used in different places (explorations and questions) and I tried to use the same structure as it's already present there. 

We have to think about how we are going to differentiate the modal response based on the use of state editor directive (whether to save the modal response in an exploration saving method or the questions' saving method). Currently, I'm not sure how I can do this but If you would like to get this done right now then I can think about it otehrwise I can create an issue to track this and solve it later. Let me know what do you think?

Reply:
I think maybe we can make state editor accept an option for this and state editor will initiate the service based on that value? (Still, I would prefer to do this in other PR!)

Reply:
Later is fine, I think. Please file an issue and link to it from here. Thanks!

Reply:
Done! (#6197)